src/PostgREST/ContentTypes.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+
+module PostgREST.ContentTypes
```

Comment:
I think moving those `ContentType` things out to a separate module makes a lot of sense.

In terms of naming, I think the module should be named `ContentType` - so have the same name as the main data type that it exposes. It feels kind of like a "class" in OOP.

When importing, I would then suggest to follow a pattern like this:
```haskell
import PostgREST.ContentType (ContentType(..))
import PostgREST.ContentType as ContentType (toHeader, toMime, decode) -- note how `decodeContentType` is renamed, because it will be prefixed

-- and then use:
ContentType
ContentType.decode
-- etc
```

This pattern can also be used with a shortcut with some tooling:
https://theam.github.io/require/

(Although I don't know how that would play with out import tooling - probably would just need to resolve those require statements before parsing, too)

Reply:
I think it would be good to reflect the similarity with show/read of `toMime` and `decodeContentType` by renaming them to something like `toText` and `fromText`.

```haskell
ContentType.toText
ContentType.fromText
```

That would be very natural to read. Not sure, if we will be able to follow this pattern in other places, too - it might result in a lot of Modules, basically 1 for each type, if we take it to the extreme. Otherwise, we'd get name conflicts for multiple types in the same module.

Except for, when we just make `toText` and `fromText` our own typeclasses. We'd basically just give `Show`/`Read` a different name for our use-case - but avoid all the implicit and explicit conventions about `Show`/`Read`.

We would not need to follow the `read . show = id` law, but would still have a common interface. And we'd work with `Text`, not `String`.